app/Rom.php

```diff
@@ -699,6 +701,26 @@ public function write(int $offset, $data) {
 	}
 
 	/**
+	 * Apply a JSON string containing patches to the ROM
+	 *
+	 * The string should consist of an array of patches, where each patch is an object
+	 * with keys of integer addresses and values of data to write at that address in
+	 * the form of arrays of integers between 0 and 255.
+	 *
+	 * @param string $patches JSON string of patches to apply
+	 *
+	 **/
+	public function applyJSONPatches(string $json) {
+		$patches = json_decode($json, true);
+		foreach ($patches as $patch) {
+		foreach ($patch as $address => $data) {
+			$this->write(intval($address), pack('C*', ...array_values($data)));
```

Comment:
Actually, after some testing, decoding the JSON into an array is casting the addresses to `int`s already and my `intval` cast was superfluous.